apm-sniffer/bytebuddy-patch/src/test/java/org/apache/skywalking/apm/agent/bytebuddy/case1/AbstractInterceptTest.java

```diff
@@ -0,0 +1,247 @@
+                                builder = builder.visit(new SWAsmVisitorWrapper());
+                            }
+                            return builder
+                                    .method(ElementMatchers.nameContainsIgnoreCase(methodName))
+                                    .intercept(MethodDelegation.withDefaultConfiguration()
+                                            .to(new InstMethodsInter(interceptorClassName, classLoader), fieldName))
+                                    ;
+                        }
+                )
+                .with(new AgentBuilder.Listener.Adapter() {
+                    @Override
+                    public void onError(String typeName, ClassLoader classLoader, JavaModule module, boolean loaded, Throwable throwable) {
+                        System.err.println(String.format("Transform Error: interceptorClassName: %s, typeName: %s, classLoader: %s, module: %s, loaded: %s", interceptorClassName, typeName, classLoader, module, loaded));
+                        throwable.printStackTrace();
+                    }
+                })
+                .installOn(ByteBuddyAgent.install());
+    }
+
+    protected void installMethodInterceptor2(String className, String methodName, int round) {
+        String interceptorClassName = METHOD_INTERCEPTOR_CLASS + "$" + methodName + "$" + round;
+        String nameTrait = getNameTrait(round);
+        String fieldName = nameTrait + "_delegate$" + methodName + round;
+
+        AgentBuilder agentBuilder = newAgentBuilder(nameTrait);
+        agentBuilder.type(ElementMatchers.named(className))
+                .transform((builder, typeDescription, classLoader, module, protectionDomain) -> {
+                            if (deleteDuplicatedFields) {
+                                builder = builder.visit(new SWAsmVisitorWrapper());
+                            }
+                            return builder
+                                    .method(ElementMatchers.nameContainsIgnoreCase(methodName))
+                                    .intercept(Advice.to(InstMethodAdvice.class));
+                        }
+                )
+                .with(new AgentBuilder.Listener.Adapter() {
+                    @Override
+                    public void onError(String typeName, ClassLoader classLoader, JavaModule module, boolean loaded, Throwable throwable) {
+                        System.err.println(String.format("Transform Error: interceptorClassName: %s, typeName: %s, classLoader: %s, module: %s, loaded: %s", interceptorClassName, typeName, classLoader, module, loaded));
+                        throwable.printStackTrace();
+                    }
+                })
+                .installOn(ByteBuddyAgent.install());
+    }
+
+    protected void installConstructorInterceptor(String className, int round) {
+        installConstructorInterceptor1(className, round);
+//        installConstructorInterceptor2(className, round);
```

Comment:
Forget to remove? What is intercetopro1 and interceptor2? Could you give a more proper name?